scopes/react/component-highlighter/hover-highlighter/bubble-to-component.tsx

```diff
@@ -0,0 +1,69 @@
+import { domToReacts, toRootElement } from '@teambit/react.modules.dom-to-react';
+import {
+  componentMetaField,
+  hasComponentMeta,
+  ReactComponentMetaHolder,
+} from '@teambit/react.ui.highlighter.component-metadata.bit-component-meta';
+import { ruleMatcher, MatchRule, ComponentMatchRule, componentRuleMatcher } from '../rule-matcher';
+
+type BubblingOptions = {
+  /** filter elements by this rule */
+  elementRule?: MatchRule;
+  /** filter components by this rule */
+  componentRule?: ComponentMatchRule;
+  /**
+   * continue bubbling when encountering a parent of the same component
+   * @default true
+   */
+  propagateSameParents?: boolean;
+};
+
+/** go up the dom tree until reaching a react bit component */
+export function bubbleToComponent(
+  element: HTMLElement | null,
+  { elementRule, componentRule, propagateSameParents = true }: BubblingOptions = {}
+) {
+  let current = bubbleToFirstComponent(element, elementRule, componentRule);
+  if (!propagateSameParents) return current;
+
+  while (current) {
+    const parentElement = current.element.parentElement;
+    const parent = bubbleToFirstComponent(parentElement, elementRule, componentRule);
+
+    const primeComponent = current?.components.slice(-1).pop();
+    const parentPrimeComponent = parent?.components.slice(-1).pop();
+
+    if (primeComponent !== parentPrimeComponent) return current;
```

Comment:
could compare string ids like so - `primeComponent.id !== parentPrimeComponent.id`
it will allow the highlighter to bubble to different **React Components** under the same **Bit Component**